test/new-e2e/tests/agent-subcommands/hostname/hostname_ec2_nix_test.go

```diff
@@ -47,13 +46,13 @@ func (v *linuxHostnameSuite) TestAgentConfigPreferImdsv2() {
 }
 
 // https://github.com/DataDog/datadog-agent/blob/main/pkg/util/hostname/README.md#the-current-logic
-func (v *linuxHostnameSuite) TestAgentHostnameDefaultsToResourceId() {
-	v.UpdateEnv(awshost.ProvisionerNoFakeIntake(v.GetOs(), awshost.WithAgentOptions(agentparams.WithAgentConfig(""))))
+func (v *linuxHostnameSuite) TestAgentHostnameDefaultsToNotInstanceId() {
+	v.UpdateEnv(awshost.ProvisionerNoFakeIntake(v.GetOs(), awshost.WithAgentOptions(agentparams.WithAgentConfig("ec2_prefer_imdsv2: false"))))
 
 	metadata := client.NewEC2Metadata(v.T(), v.Env().RemoteHost.Host, v.Env().RemoteHost.OSFamily)
 	hostname := v.Env().Agent.Client.Hostname()
 
-	// Default configuration of hostname for EC2 instances is the resource-id
-	resourceID := metadata.Get("instance-id")
-	assert.Equal(v.T(), resourceID, hostname)
+	instanceID := metadata.Get("instance-id")
+	// As IMDSv1 is disabled by default, the hostname should not be the instance ID
+	assert.NotEqual(v.T(), instanceID, hostname)
```

Comment:
I don't think we want to test that it doesn't default to something, that doesn't seem very useful 😄
Would you be able to update the test to make sure it defaults to whatever we expect it to ?